src/pylir/LLVM/PylirGC.cpp

```diff
@@ -133,6 +133,7 @@ class PylirGCMetaDataPrinter final : public llvm::GCMetadataPrinter
 
         for (auto& iter : callSiteInfos)
         {
+            os.emitValueToAlignment(4);
```

Comment:
Maybe not needed

Reply:
I don't think that is needed, since the code that reads this structure uses `memcpy` to copy these into aligned memory.

Reply:
Removing this line caused errors from the linker, I think Apple ARM64 cannot do unaligned access at all

Reply:
Ahh I see. So the linker can't handle relocations if they're not aligned properly. I did not consider that and only thought it was about the reading code (which circumvents that restriction via `memcpy`). 

I am wondering now however why the value 4? I'd expect it to require alignment equal to `pointerSize`. 
The code reading the stackmap: https://github.com/zero9178/Pylir/blob/6ec52b054ef4960931ead397557e9c31d80c214d/src/pylir/Runtime/Stack.cpp#L73
will also need adjustments to align `curr` to `alignof(void*)` before reading `programCounter`. 
I hope/am guessing that is the reason the Linux builds failed.

Reply:
ARM64 is 4 byte aligned so I just used 4 lol, should be the current platform alignment.

I don't currently have b/w to look at the stackmap reader though, have tons of stuff on hand

Reply:
No worries! I think I should be able to push to your branch as well so I'll just do it once I got time as well to make sure there are no regressions so you can continue working on the mac specifics.